File: openapi-generator/dart/lib/model/inline_object22.dart
part of openapi.api;

class InlineObject22 {
  /* Group ID to query. */
  int groupId = null;
  /* List of sensor IDs to query. */
  List<int> sensors = [];
  InlineObject22();

  @override
  String toString() {
    return 'InlineObject22[groupId=$groupId, sensors=$sensors, ]';
  }

  InlineObject22.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['groupId'] == null) {
      groupId = null;
    } else {
          groupId = json['groupId'];
    }
    if (json['sensors'] == null) {
      sensors = null;
    } else {
      sensors = (json['sensors'] as List).cast<int>();
    }
  }

  Map<String, dynamic> toJson() {
    return {
      'groupId': groupId,
      'sensors': sensors
    };
  }

  static List<InlineObject22> listFromJson(List<dynamic> json) {
    return json == null ? new List<InlineObject22>() : json.map((value) => new InlineObject22.fromJson(value)).toList();
  }

  static Map<String, InlineObject22> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, InlineObject22>();
    if (json != null && json.length > 0) {
      json.forEach((String key, dynamic value) => map[key] = new InlineObject22.fromJson(value));
    }
    return map;
  }
}

File: openapi-generator/scalaz/src/main/scala/org/openapitools/client/api/InlineObject22.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime
import InlineObject22._

case class InlineObject22 (
  /* Group ID to query. */
  groupId: Long,
/* List of sensor IDs to query. */
  sensors: List[Long])

object InlineObject22 {
  import DateTimeCodecs._

  implicit val InlineObject22CodecJson: CodecJson[InlineObject22] = CodecJson.derive[InlineObject22]
  implicit val InlineObject22Decoder: EntityDecoder[InlineObject22] = jsonOf[InlineObject22]
  implicit val InlineObject22Encoder: EntityEncoder[InlineObject22] = jsonEncoderOf[InlineObject22]
}
